tiledb/sm/c_api/tiledb.cc

```diff
@@ -5908,6 +5910,103 @@ int32_t tiledb_deserialize_array_schema(
   return TILEDB_OK;
 }
 
+int32_t tiledb_serialize_array_open(
+    tiledb_ctx_t* ctx,
+    const tiledb_array_t* array,
+    tiledb_serialization_type_t serialize_type,
+    int32_t client_side,
+    tiledb_buffer_t** buffer) {
+  // client_side is currently unused
```

Comment:
this could use a sentence or two about what it _will_ be used for when it is used